### PR TITLE
Merge win cumu

### PR DESCRIPTION
**Pull request by JeGoi**
Merge packetfence ansible roles:
- https://github.com/inverse-inc/ansible-windows
- https://github.com/inverse-inc/ansible-cumulus
- https://github.com/inverse-inc/ansible-packetfence (wpasupplicant)
It will help to centralise roles for automation tests

### CODE REVIEW

**Comment by JeGoi** [comment]
A silent modification in wpa_supplicant role for multiple templates.
Need to be test/modified in current workflow.

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (nqb) [inline LICENSE:1]: We don't need this file because `galaxy.yml` already contains license informations.

File: playbooks/service/venom.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
```
Comment (nqb) [inline playbooks/service/venom.yml:1]: This file can be removed.

File: README.md [hunk 3]
```diff
@@ -1,3 +1,8 @@
 # Ansible Collection - inverse_inc.utils
 
-Collection to install utils on PacketFence servers.
+- Collection to install utils on PacketFence servers.
+- Collection to manage softwares running on Cumulus Linux.
+- Collection to be used for venom testing purposes.
```
Comment (nqb) [inline README.md:5]:
```suggestion
- Collection to be used for Venom testing purposes.
```

File: roles/hostapd/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    - name: Ensure kernel installed
+      include_tasks: w_kernel.yml
+
+    - name: Ensure hostapd configured installed
+      include_tasks: w_configure.yml
+
+    - name: Ensure hostapd services set
+      include_tasks: w_services.yml
+
+  when: "'wireless' in inventory_hostname"
```
Comment (nqb) [inline roles/hostapd/tasks/main.yml:17]: This conditional make role unusable unless you have a wireless device in inventory.

File: roles/hostapd/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        dest: /etc/hostapd.conf
+      notify: apply changes
+
+    - name: Start and enable hostapd service
+      service:
+        name: hostapd
+        state: "{{ hostapd__state }}"
+        enabled: "{{ hostapd__enabled }}"
+
+  when: "'switch01' in inventory_hostname"
```
Comment (nqb) [inline roles/hostapd/tasks/main.yml:34]: This conditional make role unusable unless you have a switch01 device in inventory.

File: roles/hostapd/templates/hostapd_default.conf.j2 [hunk 6]
```diff
@@ -0,0 +1,2888 @@
+##### hostapd configuration file ##############################################
```
Comment (nqb) [inline roles/hostapd/templates/hostapd_default.conf.j2:1]: I don't think this file is necessary.

File: roles/hostapd/templates/hostapd_kernel_boot.conf.j2 [hunk 7]
```diff
@@ -0,0 +1 @@
+mac80211_hwsim
```
Comment (nqb) [inline roles/hostapd/templates/hostapd_kernel_boot.conf.j2:1]: This is not a template.

File: roles/hostapd/templates/hostapd.service.j2 [hunk 8]
```diff
@@ -0,0 +1,17 @@
+[Unit]
+Description=Hostapd {{ item }}       IEEE 802.11 AP, IEEE 802.1X/WPA/WPA2/EAP/RADIUS Authenticator
+Conflicts=hostapd.service {% if item == "open" %}hostapd_secure{% else %}hostapd_open{% endif %}.service
+After=network.target hostapd.service {% if item == "open" %}hostapd_secure.service{% endif %}
```
Comment (nqb) [inline roles/hostapd/templates/hostapd.service.j2:4]: I'm not sure assignment based on test is correct (item=open, assigne hostapd_secure)

**Review by nqb — CHANGES_REQUESTED** [review]
1. Could you describe why you want to merge Windows roles into ansible-utils ?

I don't see a real benefit of that.

2. Regarding usage of `hostapd` role for Cumulus Linux on our VM, I think it add complexity and conditional in the code. Again, I don't see a real benefit of that.

Having two different roles will be more clear and avoid conditionals.

3. Regarding management of Jinja templates:
- w_hostapd.conf.j2
- wpa_supplicant.conf.j2

I think we should directly convert YAML dict into key=value assignment in order to remove repetitive test.

4. Defaults values: I'm not sure it's necessary to add open and secure definitions in defaults/main.yml. 

We want that but we need to define this in PacketFence source tree.

File: roles/hostapd/templates/hostapd.service.j2 (hunk 8 above)
Comment (JeGoi) [reply roles/hostapd/templates/hostapd.service.j2:4]: Yes, it supposed to stop the hostapd.service and hostapd_secure.service (avoid conflict).
In this role, you have three services:
- hostapd.service
- hostapd_open.service
- hostapd_secure.service
When one is started, the two others need to be stopped.

The main idea was to use `service hostapd_secure start` to stop `hostapd.service` and `hostapd_open.service` if one of them is running. Same for `service hostapd_open start` to stop `hostapd.service` and `hostapd_secure.service` if one of them is running.
Another way is to stop, unwanted services and start the wanted service.

File: roles/hostapd/templates/hostapd_kernel_boot.conf.j2 (hunk 7 above)
Comment (JeGoi) [reply roles/hostapd/templates/hostapd_kernel_boot.conf.j2:1]: Right it could be moved in file.

File: roles/hostapd/tasks/main.yml (hunk 4 above)
Comment (JeGoi) [reply roles/hostapd/tasks/main.yml:17]: Yes that is right. It was to make a difference between wired and wireless configuration.
Same for your next point.
But there is probably a better solution. Create a service for the wired with the same way as the open and secure wireless.
To be discuss.

File: roles/hostapd/templates/hostapd_kernel_boot.conf.j2 (hunk 7 above)
Comment (JeGoi) [reply roles/hostapd/templates/hostapd_kernel_boot.conf.j2:1]: Right, it should be moved in ./files/

**Comment by JeGoi** [comment]
>     1. Could you describe why you want to merge Windows roles into ansible-utils ?
> 
> 
> I don't see a real benefit of that.
> 

To have only one ansible-utils for testing and not many ansible repo that includes ansible-windows, ansible-cumulus, part of ansible-packetfence
The other question could be: why a specific ansible-windows? After that merge, only ansible-utils will be used for testing purposes.
To be discuss.

>     2. Regarding usage of `hostapd` role for Cumulus Linux on our VM, I think it add complexity and conditional in the code. Again, I don't see a real benefit of that.
> 

For this specific role, the main goal was to merge wired and wireless without breaking anything. The conditional could be remove and a wired service could be set for wired tests or set as default config like it is currently done.

> 
> Having two different roles will be more clear and avoid conditionals.
>

Hum, not so sure about that. I guess we can do both without conditional and without modifications on Venom.
I will try.

>     3. Regarding management of Jinja templates:
> 
> 
>     * w_hostapd.conf.j2
> 
>     * wpa_supplicant.conf.j2
> 
> 
> I think we should directly convert YAML dict into key=value assignment in order to remove repetitive test.

These tests allow us to create multiples options.
Not have extra line in config it is only manage in the dict so transparent for the user.
To be discuss.

> 
>     4. Defaults values: I'm not sure it's necessary to add open and secure definitions in defaults/main.yml.
> 
> 
> We want that but we need to define this in PacketFence source tree.

Ok It will be removed from the default/main.yml

File: roles/hostapd/templates/hostapd.service.j2 (hunk 8 above)
Comment (JeGoi) [reply roles/hostapd/templates/hostapd.service.j2:4]: ~~There is a circle around this part.
If you start on, you need to stop the other, but at one point, you need to exit from that circle. The `After=` with only `hostapd_secure.service` on `hostapd_open.service`.~~

It looks that Conflicts= is enough...

**Pull request closed without merging**